src/test/java/org/sonarsource/sonarlint/ls/mediumtests/LanguageServerMediumTests.java

```diff
@@ -331,12 +330,37 @@ void noAnalysisOnNullContent() throws Exception {
     lsProxy.getTextDocumentService()
       .didClose(new DidCloseTextDocumentParams(docId));
 
+    assertThat(client.getDiagnostics(uri)).isNull();
+  }
+
+  @Test
+  void vcsIgnoredShouldNotAnalyzed() throws Exception {
+    emulateConfigurationChangeOnClient("**/*Test.js", true, true, true);
+    Thread.sleep(1000);
+    client.logs.clear();
+
+    String uri = getUri("foo.py");
+    client.diagnosticsLatch = new CountDownLatch(1);
+    client.isIgnoredByScm = true;
+    VersionedTextDocumentIdentifier docId = new VersionedTextDocumentIdentifier(uri, 1);
+
+    // SLVSCODE-157 - Open/Close/Open/Close triggers a race condition that nullifies content
```

Comment:
I think you can remove the 2 consecutives open/close, this is not what you want to test here.

Reply:
```suggestion
```